spec/v1/signature/jws.go

```diff
@@ -0,0 +1,56 @@
+package signature
```

Comment:
> It is strange to have spec/v1 versioning inside notation-go as it is already versioned in SemVer2.

Adding a folder for each spec version I'm decoupling the versioning of `notation-go` with the versioning of the spec. With this I want to allow evolving other packages within `notation-go`, i.e. `signature/jws` and `plugin/manager`, without them being constrained by the spec version. In other words, what if we want to redesign `signature/jws` in a way that is not backwards compatible? We would have to bump `notation-go` major version, but we couldn't because the major version would refer to the spec, not to the library.

> Also, what if plugin and signature are versioned differently?

The plugin spec references the signature spec in multiple places, e..g.: supported envelopes, supported key types, supported hash algorithms, certificate requirements, etc. Given this fact, for me it makes more sense to version them together, else we would have to think very carefully how we deal with users using different versions of each spec.

Yet, I'm open to other opinions!

Reply:
In fact, I took the idea from https://github.com/opencontainers/image-spec/tree/main/specs-go.

Reply:
Here I think we pop up an interesting question. Does `notation-go` implementation support multiple specs at the same time?

It makes sense for an implementation to support multiple versions of protocols for compatibility purposes. Does it make sense to support an old signature spec if we have a new one?

In addition, spec changes are considered as breaking changes. Therefore, the major version should be bumped up. In other words, `github.com/notaryproject/notation-go` will become `github.com/notaryproject/notation-go/v2`, and adopters can import both packages at the same time for supporting both versions of specs.

Reply:
I see your point. Then we would be bumping `notation-go` major version when there is a new spec major version or when there is a breaking change in its Go API. So it might be that `notation-go/v4` implements, i.e., the notary spec v3.

Works for me, I'll remove the `v1` folder.

Reply:
@qmuntal , @shizhMSFT adding some info to help with deciding the best course of action, there are multiple versions to deal with here. 
* Notary core specification include - signature specifications, supported envelopes, supported key types and algorithms, and cert requirements. These can be extended or some constraints may be relaxed, which would be non breaking changes.
* Breaking changes to Notary core specs would require major version bump.
* Trust policy language (`trustPolicy.json`) has it's own versioning, `notation-go` can support multiple major or minor versions of trust policy format, as long as the changes are compatible with Notary core specs.
* Plugin contract has it's own versioning, major and minor versions of plugin contract introduce newer `capabilities` and corresponding newer plugin commands. Check `supported-contract-versions` in [plugin spec](https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md#user-content-plugin-metadata).  `notation-go` is expected to support at least multiple minor versions of plugin contract (and possibly major version if we want to), existing plugin vendors should not be forced to re-implement and distribute when new minor version of plugin contract spec is published, and then implemented by `notation-go`.

Reply:
I agree with @shizhMSFT that it would be best to avoid utilizing the `v1` folder format and instead rely on versioning via the normal GitHub SemVer release process.

If we remove the `v1` folder, SemVer on release process for notation-go, and then SemVer release inside the plugin code itself (on separate repos) - will this satisfy the points you're asking @gokarnm?

Reply:
I've removed the `v1` folder. It would make sense if everything is versioned together, but if every spec has its own lifecycle, then we will have to find another strategy for `notation-go` to support multiple spec version.
I don't think GitHub releases are the solution, as `notation-go` implements several independent specs (plugins, signature, trust store, etc) and we can just say notation-go@v1 only works with this and that fixed versions of the other specs. We can't forsee which version combinations will be used in the wild, so we have to provide a flexible interface.

Anyway, this goes out of the scope for this PR.